### DISCUSSION TITLE
It doesn't work of GrpcChannelOptions.MaxReceiveMessageSize

### DISCUSSION THREAD

**Discussion by karlwang1983** [authzed/spicedb#1965] [Q&A]
We build a dotnet client following the page's guide -- https://medium.com/kpmg-uk-engineering/getting-started-with-spicedb-in-net-741e353a4d83. It works very well except GrpcChannelOptions.MaxReceiveMessageSize. 

We noticed that there are some grpc error "grpc: received message after decompression larger than max (4194305 vs. 4194304)". We know it caused by default size is 4M for grpc. Then we set GrpcChannelOptions.MaxReceiveMessageSize as below to 50M, but it still doesn't work. Anyone know how to make it works?

`           
            void ConfigureSpiceDbGrpcClient(GrpcClientFactoryOptions options)
            {
                options.Address = new Uri(settings.SpiceDbServer);
            }

            void ConfigureSpiceDbChannel(GrpcChannelOptions options)
            {
                var credentials = CallCredentials.FromInterceptor((context, metadata) =>
                {
                    metadata.Add("Authorization", $"Bearer {settings.SpiceDbToken}");
                    return Task.CompletedTask;
                });

                options.MaxReceiveMessageSize = 50 * 1024 * 1024;
                options.UnsafeUseInsecureChannelCallCredentials = true;
                options.Credentials = ChannelCredentials.Create(ChannelCredentials.Insecure, credentials);
            }

            services.AddGrpcClient<PermissionsService.PermissionsServiceClient>(ConfigureSpiceDbGrpcClient)
                .ConfigureChannel(ConfigureSpiceDbChannel);`

**Comment by vroldanbet** [comment]
@ karlwang1983 I'm not well versed on dotnet, so I don't comment on any specifics of the generated SDK, but I would like to know which API request are you triggering to hit this. Can you reproduce this if you use `zed` command line?

**Reply by karlwang1983** [reply]
LookSubjects API. The total size of subjects exceed the grpc max received size. That's why I try to use cursor.  I can also use python client but I don't know how to set this value in python. Do you have any idea about set max receive size in python client? @vroldanbet

**Reply by vroldanbet** [reply]
I'm sorry but I don't know and don't have the bandwidth to look into this right now. Maybe someone else in the community can chime in. That's also why I suggested using `zed` to do the lookup subjects, to rule out language SDK specifics.
